sdv/metadata/metadata.py

```diff
@@ -324,6 +324,32 @@ def add_column(self, column_name, table_name=None, **kwargs):
         table_name = self._handle_table_name(table_name)
         super().add_column(table_name, column_name, **kwargs)
 
+    def remove_table(self, table_name):
+        """Remove a table from the metadata.
+
+        This method removes a table from the metadata as well as all relationships that table
+        contains.
+
+        Args:
+            table_name (str):
+                The name of the table to remove.
+        """
+        self._validate_table_exists(table_name)
+
+        # Remove relationships
+        updated_relationships = []
+        for relationship in self.relationships:
+            relationship_tables = (
+                relationship['parent_table_name'],
+                relationship['child_table_name'],
+            )
+            if table_name not in relationship_tables:
+                updated_relationships.append(relationship)
```

Comment:
```suggestion
        updated_relationships = []
        for relationship in self.relationships:
            relationship_tables = (
                relationship['parent_table_name'],
                relationship['child_table_name'],
            )
            if table_name in relationship_tables:
                self.remove_relationship(relationship['parent_table_name'], relationship['child_table_name'])

Reply:
I can't do this because you can remove elements from a list as you loop through it

Reply:
I think this would be less efficient. That method is meant to remove a relationship given a parent and child and loops through all the relationships to do it. This method is meant to remove relationships only based on one table

Reply:
@pvk-developer I'll refactor to re-use the shared parts